docs/Modeling-Artifacts.md

```diff
@@ -0,0 +1,367 @@
+# Modeling Artifacts
+
```

Comment:
For better visibility, should we publish this doc to the ORAS website, e.g. https://oras.land/docs/client_libraries/go?

Reply:
I think we can put a link on the ORAS website and point to this? So that we only need to maintain a single copy.

Reply:
Yeah I think it makes more sense to keep it in the `oras-go` repo